apps/dotcdn/src/app/dotcdn.component.store.spec.ts

```diff
@@ -0,0 +1,176 @@
+
+const fakeStateViewModel = {
+    chartBandwidthData: {
+        labels: ['15/04', '16/04', '17/04', '18/04', '19/04', '20/04', '21/04', '22/04', '23/04'],
+        datasets: [
+            {
+                label: 'Bandwidth Used',
+                data: ['78.55', '15.51', '0.00', '0.00', '15.30', '5.06', '0.00', '0.00', '0.00'],
+                borderColor: '#6f5fa3',
+                fill: false
+            }
+        ]
+    },
+    chartRequestsData: {
+        labels: ['15/04', '16/04', '17/04', '18/04', '19/04', '20/04', '21/04', '22/04', '23/04'],
+        datasets: [
+            {
+                label: 'Requests Served',
+                data: ['78554075', '15506849', '0', '0', '15301470', '5061682', '0', '0', '0'],
+                borderColor: '#FFA726',
+                fill: false
+            }
+        ]
+    },
+    statsData: [
+        {
+            label: 'Bandwidth Used',
+            value: '114.42 MB',
+            icon: 'insert_chart_outlined'
+        },
+        {
+            label: 'Requests Served',
+            value: '130',
+            icon: 'file_download'
+        },
+        {
+            label: 'Cache Hit Rate',
+            value: '14.62%',
+            icon: 'file_download'
+        }
+    ],
+    isChartLoading: false,
+    cdnDomain: 'demo.dotcms.com'
+};
+
+describe('DotCDNComponentStore', () => {
+    let store: DotCDNStore;
+    let dotCdnService: DotCDNService;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [HttpClientTestingModule],
+            providers: [
+                DotCDNService,
+                DotCDNStore,
+                { provide: CoreWebService, useClass: CoreWebServiceMock },
+                { provide: SiteService, useClass: SiteServiceMock }
+            ]
+        });
+        store = TestBed.inject(DotCDNStore);
+        dotCdnService = TestBed.inject(DotCDNService);
+    });
+
+    describe('DotCDN Component Store', () => {
+        beforeEach(() => {
+            jest.restoreAllMocks();
+            jest.clearAllMocks();
+        });
+
+        it('should set chart state', (done) => {
+            jest.spyOn(dotCdnService, 'requestStats').mockReturnValue(of(fakeResponseData));
+
+            store.getChartStats('30');
+
+            store.vm$.subscribe((state) => {
+                expect(state).toStrictEqual(fakeStateViewModel);
+                done();
+            });
+        });
+
+        it('should purge cdn with urls', (done) => {
+            const urls = ['url1, url2'];
+
+            const dotCdnServiceSpy = jest.spyOn(dotCdnService, 'purgeCache').mockReturnValue(
+                of({
+                    entity: {
+                        'All Urls Sent Purged: ': true
+                    },
+                    errors: [],
+                    i18nMessagesMap: {},
+                    messages: [],
+                    permissions: []
+                })
+            );
+
+            store.purgeCDNCache(urls);
```

Comment:
This method mutates the state twice because if loading and you are not asserting that.